restapi.go

```diff
@@ -2533,6 +2533,12 @@ func (s *Session) ApplicationCommands(appID, guildID string) (cmd []*Application
 func (s *Session) InteractionRespond(interaction *Interaction, resp *InteractionResponse) error {
 	endpoint := EndpointInteractionResponse(interaction.ID, interaction.Token)
 
+	if resp.Data != nil {
+		if resp.Data.Ephemeral {
+			resp.Data.Flags = 1 << 6
```

Comment:
i think you can have ephemeral with other types of flags in theory. (although i don't think any other flags are supported currently) im thinking the proper way to do this might be `resp.Data.Flags = resp.Data.Flags | (1 << 6)`